File: server/BooksStorage/Models/Book/Book.cs
using System.ComponentModel.DataAnnotations;
using MongoDB.Bson;
using MongoDB.Bson.Serialization.Attributes;

namespace BooksStorage.Models.Book;

public class Book
{
    [BsonId] public ObjectId Id { get; set; }

    [Required] public string Name { get; set; }
    [Required] public string Author { get; set; }

    [Required]
    [Display(Name = "Year of Publication")]
    public int PublicationYear { get; set; }

    [Required] public decimal Price { get; set; }
    [Required] public string Category { get; set; }

    public long ViewCount { get; set; } = 0;
}

File: server/booksstorage/internal/models/book.go
package models

type Book struct {
	Name     string  `bson:"name,omitempty"`
	Author   string  `bson:"author,omitempty"`
	PubYear  int     `bson:"publication_year,omitempty"`
	Price    float64 `bson:"price,omitempty"`
	Category string  `bson:"category,omitempty"`
}
